src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -48,7 +60,8 @@
     (log/warn "Cannot save QueryExecution, missing :context")
     (let [qe-id (t2/insert-returning-pk! QueryExecution (dissoc query-execution :json_query))]
       (when (seq field-usages)
-        (t2/insert! :model/FieldUsage (map #(assoc % :query_execution_id qe-id) field-usages))))))
+        (doseq [field-usage (map #(assoc % :query_execution_id qe-id) field-usages)]
+          (grouper/submit! @field-usages-queue field-usage))))))
```

Comment:
nit: I think it'll be syntactically clearer to do the `assoc` in the body of the `doseq` rather than using map. It also saves an extra layer of iteration. Dereferencing the delay also has so overhead so I would move that out of the loop.

```suggestion
(let [queue @field-usages-queue]
  (doseq [fu field-usages]
    (grouper/submit! queue (assoc fu :query_execution_id qe-id))))))))
```